src/common/components/header/index.tsx

```diff
@@ -0,0 +1,82 @@
+import styled from 'styled-components';
+
+import { GoBackIcon } from '@/assets/images/svgs';
+import { HeaderType } from '@/common/components/header/index.type';
+
+function Header({ title, showIcon, type }: HeaderType) {
+  return (
+    <>
+      {type === 1 && ( //type === 1 : title이 왼쪽에 있는경우
+        <HeaderArea>
+          {showIcon ? <GoBackIcon /> : <DummyIcon />}
+          <TitleArea>{title}</TitleArea>
+        </HeaderArea>
+      )}
+
+      {type === 2 && ( //type === 2 : title이 정중앙에 있는 경우
+        <HeaderArea2>
+          {showIcon ? <GoBackIcon /> : <DummyIcon />}
+          <TitleArea2>{title}</TitleArea2>
+        </HeaderArea2>
+      )}
```

Comment:
여기 분기처리를 조금 더 깔끔하고 재사용성 있게 개선할 수 있을 것 같아요 !

**1. ts-pattern 사용**
이렇게 깔끔하게 분기처리를 할 수 있어서 저는 ts-pattern을 즐겨써요! 

```
    match(type)
    .with(1, () => (
      <HeaderArea>
        {showIcon ? <GoBackIcon /> : <DummyIcon />}
        <TitleArea>{title}</TitleArea>
      </HeaderArea>
    ))
    .with(2, () => (
      <HeaderArea2>
        {showIcon ? <GoBackIcon /> : <DummyIcon />}
        <TitleArea2>{title}</TitleArea2>
      </HeaderArea2>
    ))
    .otherwise(() => null);
};
```
[ts-pattern을 활용하여 해결했던 것](https://www.kimcoder.io/blog/ts-pattern#%EC%A3%BC%EC%9A%94-API)

**2. if-else/swtich-case문 사용**
최근 ts-pattern의 match는 성능적으로 if-else / switch 문에 비해 많이 불리하다는 단점이 있다는 것을 알게되었어요. 그래서 지금처럼 단순 분기처리의 경우 ts-pattern을 사용하지 않고 switch case를 사용해 컴포넌트를 렌더링해주는 게 더 적절할 것 같아요 !

[ts-pattern은 더 멋진 if문이 아니다
](https://toss.tech/article/ts-pattern-usage)

어떤 기술을 선택할 지는 영수님 마음~~

```suggestion
const renderHeaderContent = (type: number, showIcon: boolean, title: string) => {
      switch (type) {
        case 1:
          return (
            <HeaderArea>
              {showIcon ? <GoBackIcon /> : <DummyIcon />}
              <TitleArea>{title}</TitleArea>
            </HeaderArea>
          );
        case 2:
          return (
            <HeaderArea2>
              {showIcon ? <GoBackIcon /> : <DummyIcon />}
              <TitleArea2>{title}</TitleArea2>
            </HeaderArea2>
          );
        default:
          return null;
      }
};

function Header({ title, showIcon, type }: HeaderType)  {
     return <>{renderHeaderContent(type, showIcon, title)}</>;
};
```

Reply:
엄청 세세하게 봐주셔서 감사합니다. ts-pattern에 대해 처음 들어 봤는데 이번에 한번 공부하고 적극적으로 사용해볼게요 !